backends/s3/marker.go

```diff
@@ -17,19 +16,16 @@ func (b *Backend) setMarker(ctx context.Context, name, etag string, isDel bool)
 	if !b.opt.UseUpdateMarker {
 		return nil
 	}
-	ts := strconv.FormatInt(time.Now().UnixNano(), 10)
-	lines := []string{name, etag, ts}
-	if isDel {
-    		lines = append(lines, "del")
-	}
-	joined := []byte(strings.Join(lines, "\x00"))
-	_, err := b.doStore(ctx, UpdateMarkerFilename, joined)
+	nanos := time.Now().UnixNano()
+	s := fmt.Sprintf("%s\x00%s\x00%d\x00%v", name, etag, nanos, isDel)
```

Comment:
I would prefer a readable string as marker, all the fields in here are readable strings.

```suggestion
	s := fmt.Sprintf("%s:%s:%d:%v", name, etag, nanos, isDel)
```